src/index.ts

```diff
@@ -509,17 +487,98 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
       }
     }
 
-    eventSource.addEventListener('*', (msg: any) => {
-      const event: StreamEvent = JSON.parse(msg.data)
+    const processData = (data: any): void => {
+      const lines = data.toString().split(/\r?\n/);
+      lines.forEach((line) => processLine(line));
+    }
 
-      logDebug('Received event from stream: ', event)
+    const processLine = (line: string): void => {
+      if (line.startsWith('data:')) {
+        const event: StreamEvent = JSON.parse(line.substring(5))
+        logDebug('Received event from stream: ', event)
 
-      if (event.domain === 'flag') {
-        handleFlagEvent(event)
-      } else if (event.domain === 'target-segment') {
-        handleSegmentEvent(event)
+        if (event.domain === 'flag') {
+          handleFlagEvent(event)
+        } else if (event.domain === 'target-segment') {
+          handleSegmentEvent(event)
+        }
       }
-    })
+    }
+
+    const getRandom = (min, max) => {
+      return Math.round(Math.random() * (max - min) + min)
+    }
+
+    const onConnected = () => {
+      logDebug('Stream connected')
+      eventBus.emit(Event.CONNECTED)
+    };
+
+    const onDisconnect = () => {
+      clearInterval(readTimeoutCheckerId)
+      const reconnectDelayMs = getRandom(1000, 10000)
+      logDebug('Stream disconnected, will reconnect in ' + reconnectDelayMs + 'ms')
+      eventBus.emit(Event.DISCONNECTED)
+      setTimeout(() => startStream(), reconnectDelayMs)
+    };
+
+    const onFailed = (msg: string) => {
+      if (msg != null) {
+        logError('Stream has issue', msg)
+      }
+      eventBus.emit(Event.ERROR_STREAM, msg)
+      eventBus.emit(Event.ERROR, msg)
+      onDisconnect()
+    }
+
+    const url = `${configurations.baseUrl}/stream?cluster=${clusterIdentifier}`
+
+    const sseHeaders = {
+      'Cache-Control': 'no-cache',
+      'Accept': 'text/event-stream',
+      'API-Key': apiKey,
+      ...standardHeaders
+    }
+
+    logDebug('SSE HTTP start request', url);
+
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", url)
+    for (const [header, value] of Object.entries(sseHeaders)) {
+      xhr.setRequestHeader(header, value)
+    }
+    xhr.timeout = 86400000 // Force SSE to reconnect after 24hrs
+    xhr.onerror = () => { onFailed('XMLHttpRequest error on SSE stream'); }
+    xhr.onabort = () => { logDebug('SSE aborted'); onFailed(null); }
+    xhr.ontimeout = () => { onFailed('SSE timeout'); }
```

Comment:
Looks like this file could do with having `prettier` run. Inline bracketed functions shouldn't be allowed.
```suggestion
    xhr.onerror = () => {
      onFailed('XMLHttpRequest error on SSE stream');
    }
    xhr.onabort = () => {
      logDebug('SSE aborted');
      onFailed(null);
    }
    xhr.ontimeout = () => {
      onFailed('SSE timeout');
    }
```

Reply:
i agree, i will do a final pass before merging as well